src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java

```diff
@@ -495,7 +495,7 @@ public ResultSet getPrimaryKeys(final String catalog, final String schema,
         final List<List<Object>> metaData = new ArrayList<>();
         if (schema == null || properties.getDatabase().equals(schema)) {
             for (String tableName : databaseMetadata.getTableSchemaMap().keySet()) {
-                if (table == null || tableName.equals(table)) {
+                if (tableName.equals(table)) {
                     final DocumentDbSchemaTable metadataTable = databaseMetadata
```

Comment:
Should this be changed to tableName.matches(regexTablePattern)?

Reply:
I don't think so. According to the documentation, for getPrimaryKeys, the tableName should match as it is stored in the database. [reference to oracle documentation](https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html#getPrimaryKeys(java.lang.String,%20java.lang.String,%20java.lang.String))

Reply:
Ok, I see. The table name can not be a search pattern.